bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/events/BuildManager.java

```diff
@@ -739,6 +739,14 @@ public void endTopLevel(boolean needsBuild) {
 		autoBuildJob.build(needsBuild);
 	}
 
+	public void waitForBuild() {
+		autoBuildJob.wakeUp(0);
```

Comment:
The random fails may indicate that there were some tests relying on autobuild scheduling delay, or that `wakeUp(0)` has an issue (I see zero tests for that and no code at all in platform using `0` delay).